docs/rules/id-length.md

```diff
@@ -217,6 +218,29 @@ const { x } = foo;
 const { a: x } = foo;
 ```
 
+### exceptionPatterns
+
+Examples of additional **correct** code for this rule with the `{ "exceptionPatterns": ["^x_", "_send$"] }` option:
+
+```js
+/*eslint id-length: ["error", { "exceptionPatterns": ["^x_", "_send$"] }]*/
+/*eslint-env es6*/
+
+var x_value = 5;
+function longFunction_send() { return 42; }
+obj.x_longName = document.body;
+var foo = function (x_value) { /* do stuff */ };
+try {
+    dangerousStuff();
+} catch (x_value) {
+    // ignore as many do
+}
+(x_value) => {return x_value * x_value};
+var [x_value] = arr;
+const { x_value } = foo;
+const { a: x_value } = foo;
```

Comment:
Can we change examples to allowed cases affected by the option  `exceptionPatterns `?.
`x_value` seems correct example regardless of the option `exceptionPatters` option.

* [online-demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBbXCJlcnJvclwiXSovXG4vKmVzbGludC1lbnYgZXM2Ki9cbnZhciB4X3ZhbHVlID0gNTtcbmZ1bmN0aW9uIGxvbmdGdW5jdGlvbl9zZW5kKCkgeyByZXR1cm4gNDI7IH1cbm9iai54X2xvbmdOYW1lID0gZG9jdW1lbnQuYm9keTtcbnZhciBmb28gPSBmdW5jdGlvbiAoeF92YWx1ZSkgeyAvKiBkbyBzdHVmZiAqLyB9O1xudHJ5IHtcbiAgICBkYW5nZXJvdXNTdHVmZigpO1xufSBjYXRjaCAoeF92YWx1ZSkge1xuICAgIC8vIGlnbm9yZSBhcyBtYW55IGRvXG59XG4oeF92YWx1ZSkgPT4ge3JldHVybiB4X3ZhbHVlICogeF92YWx1ZX07XG52YXIgW3hfdmFsdWVdID0gYXJyO1xudmFyIHsgeF92YWx1ZSB9ID0gZm9vO1xudmFyIHsgYTogeF92YWx1ZSB9ID0gZm9vOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=) (no error)
```js
/*eslint id-length: ["error"]*/
/*eslint-env es6*/
var x_value = 5;
function longFunction_send() { return 42; }
obj.x_longName = document.body;
var foo = function (x_value) { /* do stuff */ };
try {
    dangerousStuff();
} catch (x_value) {
    // ignore as many do
}
(x_value) => {return x_value * x_value};
var [x_value] = arr;
var { x_value } = foo;
var { a: x_value } = foo;
```

Reply:
You are right. 
```longFunction_send``` isn't correct, too.
Because there is no default value for ```max-length``` in id-length rule

Maybe I thought it has a default value like 10.
Sorry for the mistake. 

Now i could think of right cases of the rule. like [a-d] or [ E | S | L | i | n | t]. 
Perhaps it will be simpler.